src/dclass.d

```diff
@@ -1139,7 +1137,10 @@ public:
                     {
                         import ddmd.access : symbolIsVisible;
 
-                        s = b.sym.search(loc, ident, flags);
+                        if (!(flags & IgnoreSymbolVisibility) && accessModule != b.sym.getAccessModule())
+                            s = b.sym.search(loc, ident, flags | IgnorePrivateImports);
```

Comment:
We should merge the old IgnorePrivateImports mechanism for imports and the new symbolIsVisible mechanism at some point.